modules/broker-auth/src/main/java/io/ballerina/messaging/broker/auth/ldap/LdapAuthHandler.java

```diff
@@ -137,12 +162,47 @@ public String searchDN(DirContext dirContext, String username) throws NamingExce
             LOGGER.debug("DN: {} obtained for Username: {}", dn, username);
         } else {
             dn = null;
-            LOGGER.debug("DN search failed for Username: {}", username);
+            LOGGER.debug("DN is not found for Username: {}", username);
         }
         answer.close();
+        closeDirContext(dirContext);
         return dn;
     }
 
+    /**
+     * Fetch the set of groups which the user belongs to.
+     *
+     * @param userDN distinguished name (dn) of the user
+     * @return a set of groups which the given user belongs to
+     * @throws NamingException in case of failing to connect to ldap server or erroneous search query
+     */
+    public Set<String> getUserGroups(String userDN) throws NamingException {
+
+        DirContext dirContext = connectAnonymously();
```

Comment:
Should this be a long-lived object?

Reply:
It will be done later as a perf improvement. If we keep the dir context this class will become stateful and the dir context should be managed in a thread safe manor.